chipsec/hal/spi.py

```diff
@@ -202,26 +230,26 @@ def __init__(self, cs):
         self.bios_ptinx  = int(self.cs.get_register_def("BIOS_PTINX")['offset'],16)
         self.bios_ptdata  = int(self.cs.get_register_def("BIOS_PTDATA")['offset'],16)
 
-        if logger().HAL:
-            logger().log( "[spi] Reading SPI flash controller registers definitions:" )
-            logger().log( "      HSFC   offset = 0x{:04X}".format(self.hsfc_off) )
-            logger().log( "      HSFS   offset = 0x{:04X}".format(self.hsfs_off) )
-            logger().log( "      FADDR  offset = 0x{:04X}".format(self.faddr_off) )
-            logger().log( "      FDATA0 offset = 0x{:04X}".format(self.fdata0_off) )
+        if self.logger.HAL:
+            self.logger.log( "[spi] Reading SPI flash controller registers definitions:" )
+            self.logger.log( "      HSFC   offset = 0x{:04X}".format(self.hsfc_off) )
+            self.logger.log( "      HSFS   offset = 0x{:04X}".format(self.hsfs_off) )
+            self.logger.log( "      FADDR  offset = 0x{:04X}".format(self.faddr_off) )
+            self.logger.log( "      FDATA0 offset = 0x{:04X}".format(self.fdata0_off) )
 
     # Fallback option when XML config is not available: using hardcoded config
-    def get_SPI_MMIO_base_fallback(self):
-        reg_value = self.cs.pci.read_dword( Cfg.SPI_MMIO_BUS, Cfg.SPI_MMIO_DEV, Cfg.SPI_MMIO_FUN, Cfg.SPI_MMIO_REG_OFFSET )
-        spi_base = ((reg_value >> Cfg.SPI_BASE_ADDR_SHIFT) << Cfg.SPI_BASE_ADDR_SHIFT) + Cfg.SPI_MMIO_BASE_OFFSET
-        if logger().HAL: logger().log( "[spi] SPI MMIO base: 0x{:016X} (assuming below 4GB)".format(spi_base) )
-        return spi_base
+    #def get_SPI_MMIO_base_fallback(self):
+    #    reg_value = self.cs.pci.read_dword( Cfg.SPI_MMIO_BUS, Cfg.SPI_MMIO_DEV, Cfg.SPI_MMIO_FUN, Cfg.SPI_MMIO_REG_OFFSET )
+    #    spi_base = ((reg_value >> Cfg.SPI_BASE_ADDR_SHIFT) << Cfg.SPI_BASE_ADDR_SHIFT) + Cfg.SPI_MMIO_BASE_OFFSET
+    #    if self.logger.HAL: self.logger.log( "[spi] SPI MMIO base: 0x{:016X} (assuming below 4GB)".format(spi_base) )
+    #    return spi_base
 
     def get_SPI_MMIO_base(self):
         if self.mmio.is_MMIO_BAR_defined('SPIBAR'):
             (spi_base,spi_size) = self.mmio.get_MMIO_BAR_base_address('SPIBAR')
-        else:
```

Comment:
if SPIBAR is not defined, i believe this function will error out because spi_base will be undefined. should either initialize spi_base and spi_size, or keep the fallback code in there i think.

Reply:
**SPIBAR** is defined in **common.xml**, we may never hit the ELSE condition.  If it does, `get_SPI_MMIO_base_fallback()` looks like a hard coded version of the **SPIBAR** def in XML.